base/constants_syncdocs.go

```diff
@@ -37,11 +40,12 @@ const (
 
 // Sync Gateway Metadata documents that should be GroupID scoped and accessed via the "WithGroupID" helper methods below
 const (
-	DCPCheckpointPrefixWithoutGroupID    = SyncDocPrefix + "dcp_ck:"   // DCPCheckpointPrefixWithoutGroupID stores a DCP checkpoint
-	SGCfgPrefixWithoutGroupID            = SyncDocPrefix + "cfg"       // SGCfgPrefixWithoutGroupID stores CfgSG/cbgt data
-	HeartbeaterPrefixWithoutGroupID      = SyncDocPrefix               // HeartbeaterPrefixWithoutGroupID stores a SG node heartbeat document
-	PersistentConfigPrefixWithoutGroupID = SyncDocPrefix + "dbconfig:" // PersistentConfigPrefixWithoutGroupID stores a database config
-	SyncFunctionKeyWithoutGroupID        = SyncDocPrefix + "syncdata"  // SyncFunctionKeyWithoutGroupID stores a copy of the Sync Function
+	DCPCheckpointPrefixWithoutGroupID       = SyncDocPrefix + "dcp_ck:"             // DCPCheckpointPrefixWithoutGroupID stores a DCP checkpoint
+	SGCfgPrefixWithoutGroupID               = SyncDocPrefix + "cfg"                 // SGCfgPrefixWithoutGroupID stores CfgSG/cbgt data
+	HeartbeaterPrefixWithoutGroupID         = SyncDocPrefix                         // HeartbeaterPrefixWithoutGroupID stores a SG node heartbeat document
+	PersistentConfigPrefixWithoutGroupID    = SyncDocPrefix + "dbconfig:"           // PersistentConfigPrefixWithoutGroupID stores a database config
+	SyncFunctionKeyWithoutGroupID           = SyncDocPrefix + "syncdata"            // SyncFunctionKeyWithoutGroupID stores a copy of the Sync Function
+	CollectionSyncFunctionKeyWithoutGroupID = SyncDocPrefix + "syncdata_collection" // CollectionSyncFunctionKeyWithoutGroupID stores a copy of the Sync Function
```

Comment:
I've been keeping `key` and `prefix` distinct in these to avoid confusion/accidental misuse and I think we should do the same here

```suggestion
	CollectionSyncFunctionPrefixWithoutGroupID = SyncDocPrefix + "syncdata_collection" // CollectionSyncFunctionPrefixWithoutGroupID stores a copy of the Sync Function
```